arangod/RestHandler/RestDocumentHandler.cpp

```diff
@@ -337,6 +337,17 @@ RestStatus RestDocumentHandler::readSingleDocument(bool generateBody) {
   OperationOptions options(_context);
   options.ignoreRevs = true;
 
+  // Check if dirty reads are allowed:
+  bool found = false;
+  std::string const& val =
+      _request->header(StaticStrings::AllowDirtyReads, found);
+  if (found && StringUtils::boolean(val)) {
+    // This will be used in `createTransaction` below, if that creates
+    // a new transaction. Otherwise, we use the default given by the
+    // existing transaction.
+    options.allowDirtyReads = true;
+  }
```

Comment:
```suggestion
  // This will be used in `createTransaction` below, if that creates
  // a new transaction. Otherwise, we use the default given by the
  // existing transaction.
  options.allowDirtyReads = _request->parsedValue(StaticStrings::AllowDirtyReads, /*default*/ false);
```